dali/operators/util/normal_distribution_op.cc

```diff
@@ -0,0 +1,105 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/util/normal_distribution_op.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+
+DALI_SCHEMA(NormalDistribution)
+                .DocStr(R"code(Creates a tensor that consists of data distributed normally.
+This operator can be ran in 3 modes, which determine the shape of the output tensor:
+1. Providing an input batch to this operator resolves in a batch of output tensors, which have the same shape as the input tensors.
+2. Providing a custom shape as an argument resolves in an output batch, where every tensor has the same (given) shape.
+3. Providing no input arguments resolves in an output batch of scalars, one per every input tensor, distributed normally.)code")
+                .NumInput(0, 1)
+                .NumOutput(detail::kNumOutputs)
+                .AddOptionalArg(detail::kMean, R"code(Mean value of the distribution)code",
+                                0.f, true)
+                .AddOptionalArg(detail::kStddev,
+                                R"code(Standard deviation of the distribution)code",
+                                1.f, true)
+                .AddOptionalArg(detail::kShape,
+                                R"code(Shape of single output tensor in a batch)code",
+                                detail::kShapeDefaultValue)
+                .AddOptionalArg(arg_names::kDtype, R"code(Data type for the output)code",
+                                DALI_FLOAT);
+
+DALI_REGISTER_OPERATOR(NormalDistribution, NormalDistributionCpu, CPU);
+
+
+#define NORM_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float16, float, double)  // NOLINT
+
+
+bool NormalDistributionCpu::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                      const workspace_t<CPUBackend> &ws) {
+  AcquireArguments(ws);
+  output_desc.resize(detail::kNumOutputs);
+  output_desc[0].shape = GetOutputShape(ws);
+  TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
+          {
+            TypeInfo type;
+            type.SetType<DType>(dtype_);
+            output_desc[0].type = type;
+          }
+  ), DALI_FAIL(make_string("Unsupported output type: ", dtype_)))  // NOLINT
+  return true;
+}
+
+
+void NormalDistributionCpu::AssignSingleValueToOutput(workspace_t<CPUBackend> &ws) {
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto &tp = ws.GetThreadPool();
+  distribution_t distribution(mean_[0], stddev_[0]);
+  TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
+          for (int sample_id = 0; sample_id < batch_size_; ++sample_id) {
+            tp.DoWorkWithID(
```

Comment:
As this fills `batch_size_` elements only I would consider if it's worth to even use the thread pool.

Reply:
Removed threadpool